ludwig/api.py

```diff
@@ -138,75 +138,91 @@ def train(
             random_seed=default_random_seed,
             debug=False,
             **kwargs
-    ):
+    ) -> Tuple[dict, Union[dict, pd.DataFrame], str]:
         """This function is used to perform a full training of the model on the
            specified dataset.
 
         # Inputs
 
-        :param dataset: (string, dict, DataFrame) source containing the entire dataset.
-               If it has a split column, it will be used for splitting (0: train,
-               1: validation, 2: test), otherwise the dataset will be randomly split.
-        :param training_set: (string, dict, DataFrame) source containing training data.
-        :param validation_set: (string, dict, DataFrame) source containing validation data.
-        :param test_set: (string, dict, DataFrame) source containing test data.
-        :param training_set_metadata: (string, dict) metadata JSON file or loaded metadata.
-               Intermediate preprocess structure containing the mappings of the input
-               CSV created the first time a CSV file is used in the same
-               directory with the same name and a '.json' extension.
-        :param data_format: (string) format to interpret data sources. Will be inferred
+        :param dataset: (Union[str, dict, pandas.DataFrame], default: `None`)
+            source containing the entire dataset to be used in the experiment.
+            If it has a split column, it will be used for splitting (0 for train,
+            1 for validation, 2 for test), otherwise the dataset will be
+            randomly split.
+        :param training_set: (Union[str, dict, pandas.DataFrame], default: `None`)
+            source containing training data.
+        :param validation_set: (Union[str, dict, pandas.DataFrame], default: `None`)
+            source containing validation data.
+        :param test_set: (Union[str, dict, pandas.DataFrame], default: `None`)
+            source containing test data.
+        :param training_set_metadata: (Union[str, dict], default: `None`)
+            metadata JSON file or loaded metadata.
+            Intermediate preprocess structure containing the mappings of the input
+            CSV created the first time a CSV file is used in the same
+            directory with the same name and a '.json' extension.
+        :param data_format: (str, default: `None`) format to interpret data sources. Will be inferred
                automatically if not specified.
```

Comment:
should add a list of supported formats: csv, tsv, parquet...

Reply:
done